cou/apps/auxiliary.py

```diff
@@ -202,17 +205,42 @@ def _get_change_require_osd_release_step(self) -> PreUpgradeStep:
         )
 
 
-@AppFactory.register_application(["ovn-central", "ovn-dedicated-chassis"])
-class OvnPrincipal(AuxiliaryApplication):
-    """Ovn principal application class."""
+class Ovn(AuxiliaryApplication):
```

Comment:
non-blocking: I personally don't like having a separate class here and do cross inheritance when it is not absolutely necessary. `app_utils` already holds helper functions for different apps, so adding another `check_ovn_version_pinning` wouldn't be so bad. And if we really want to separate the helper functions for OVN, maybe we can do what we did for `nova compute`' and create a separate utils file?

Reply:
Gabriel was doing that, but was looking strange. You can see it [here](https://github.com/canonical/charmed-openstack-upgrader/pull/332#discussion_r1538893162).

Reply:
I saw that and TBH I didn't find it weird. But if you think this is too OVN-specific, we could have it put in a separate `ovn_utils.py`.

But then again, this is a non-blocking suggestion.

Reply:
I though it was a little bit weird too because to pass the argument to the function it's verbose like `self.config["enable-version-pinning"].get("value")`. It feels more natural to me handle in the class itself, but no strong opinion.

I'm also not a big fan of `cross inheritance`, but `ovn` apps has a lot in common and the code started duplicating on two modules. Actually lint started complaining about that.

I think it's good to have one class that have share the common things of ovn. In the future if more business logic is necessary, like new sanity check or extra upgrade step changing in one place for both classes seems nice to me